### PR TITLE
[CHANGE] relabel: make the regex (.*) match new lines

### PR DESCRIPTION
**Pull request by marioferh**
Related https://github.com/prometheus/prometheus/pull/14505

### CODE REVIEW

File: model/relabel/relabel_test.go [hunk 1]
```diff
@@ -857,25 +880,6 @@ func TestConfig_UnmarshalThenMarshal(t *testing.T) {
...
-		},
-		{
-			name: "No regex provided",
-			inputYaml: `source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_port]
-separator: ;
-target_label: __meta_kubernetes_pod_container_port_number
-replacement: $1
-action: keepequal
-`,
-		},
```
Comment (bboreham) [inline model/relabel/relabel_test.go:878]: Why were these lines deleted?

**Review by jan--f — CHANGES_REQUESTED** [review]
Needs the same change in https://github.com/prometheus/prometheus/blob/0dc00825f5b0d9349320f2750a7df6699a2987c0/model/labels/regexp.go#L72 and https://github.com/prometheus/prometheus/blob/0dc00825f5b0d9349320f2750a7df6699a2987c0/promql/functions.go#L1380.
See https://github.com/prometheus/prometheus/pull/14505#issuecomment-2257087711

File: model/relabel/relabel_test.go (hunk 1 above)
Comment (marioferh) [reply model/relabel/relabel_test.go:878]: sorry it was for testing purposes, git add *

Comment (bboreham) [reply model/relabel/relabel_test.go]: Recommend you always look over what is changed in the PR, at the time you create it.
Every change should be covered by the description, either top-level or in individual commits.

**Review by jan--f — APPROVED** [review]
:+1:

File: model/relabel/relabel_test.go (hunk 1 above)
Comment (marioferh) [reply model/relabel/relabel_test.go]: thanks

File: model/labels/regexp.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
 		}
 		// Simplify the syntax tree to run faster.
 		parsed = parsed.Simplify()
-		m.re, err = regexp.Compile("^(?:" + parsed.String() + ")$")
+		m.re, err = regexp.Compile("^(?s:" + parsed.String() + ")$")
```
Comment (bboreham) [inline model/labels/regexp.go:72]: I think this is too late to add single-line mode.  The `parsed` version prior to this addition is used for analysis.
Suggests that something is lacking in unit tests, because I expect this change to break the logic.

**Review by bboreham — CHANGES_REQUESTED** [review]
Maybe also run the relevant benchmarks?  This should be a performance improvement.

**Comment by marioferh** [comment]
contine here: https://github.com/prometheus/prometheus/pull/14505
